x-pack/plugins/ml/server/lib/register_settings.ts

```diff
@@ -30,5 +34,43 @@ export function registerKibanaSettings(coreSetup: CoreSetup) {
         }),
       },
     },
+    [ANOMALY_DETECTION_ENABLE_TIME_RANGE]: {
+      name: i18n.translate('xpack.ml.advancedSettings.enableAnomalyDetectionDefaultTimeRangeName', {
+        defaultMessage: 'Enable time filter defaults for anomaly detection jobs',
+      }),
+      value: false,
+      schema: schema.boolean(),
+      description: i18n.translate(
+        'xpack.ml.advancedSettings.enableAnomalyDetectionDefaultTimeRangeDesc',
+        {
+          defaultMessage:
+            'Use the default time filter in the Single Metric Viewer and Anomaly Explorer.',
+        }
+      ),
+      category: ['Machine Learning'],
+    },
+    [ANOMALY_DETECTION_DEFAULT_TIME_RANGE]: {
+      name: i18n.translate('xpack.ml.advancedSettings.anomalyDetectionDefaultTimeRangeName', {
+        defaultMessage: 'Time filter defaults',
+      }),
+      type: 'json',
+      value: `{
+  "from": "now-15m",
+  "to": "now"
+}`,
+      description: i18n.translate(
+        'xpack.ml.advancedSettings.anomalyDetectionDefaultTimeRangeDesc',
+        {
+          defaultMessage:
+            'The time filter selection to use when viewing anomaly detection job results.',
```

Comment:
Perhaps we can make the linkage between the two settings more obvious like they did for courier:customRequestPreference and courier:setRequestPreference? For example:
```suggestion
            'The time filter selection used when ml:anomalyDetectionTimeDefaultsEnable is true.',
```